harvester_e2e_tests/integration/test_networks.py

```diff
@@ -0,0 +1,954 @@
+                        tcp, wait_timeout)
+
+        self.ssh_client(client, vip, node_user, node_password,
+                        restart_ssh, wait_timeout)
+
+        spec = api_client.vms.Spec(1, 2)
+
+        spec.user_data += cloud_user_data.format(password=vm_credential["password"])
+
+        unique_name = unique_name + "-mgmt"
+        # Create VM
+        spec.add_image(image_opensuse.name, "default/" + image_opensuse.name)
+
+        code, data = api_client.vms.create(unique_name, spec)
+        assert 201 == code, (f"Failed to create vm with error: {code}, {data}")
+
+        # Check VM start in running state
+        check_vm_running(api_client, unique_name, wait_timeout)
+
+        # Check until VM ip address exists
+        check_vm_ip_exists(api_client, unique_name, wait_timeout)
+
+        # Get VM interface ipAddresses
+        code, data = api_client.vms.get_status(unique_name)
+        assert 200 == code, (f"Failed to get specific vm content: {code}, {data}")
+
+        interfaces_data = data['status']['interfaces']
+
+        assert 1 == len(interfaces_data), (
+            f"Failed: get more than one interface: {interfaces_data}"
+        )
+
+        mgmt_ip = interfaces_data[0]['ipAddress']
+
+        # Ping management ip address from Harvester node
+        ping_command = "ping -c 50 {0}".format(mgmt_ip)
+
+        _stdout, _stderr = self.ssh_client(
+            client, vip, node_user, node_password, ping_command, wait_timeout)
+
+        stdout = _stdout.read().decode('ascii').strip("\n")
+
+        assert stdout.find(f"64 bytes from {mgmt_ip}") > 0, (
+            f"Failed to ping VM management IP {mgmt_ip} "
+            f"on management interface from Harvester node")
+
+        # SSH to management ip address and execute command from Harvester node
+        _stdout, _stderr = self.ssh_jumpstart(
+            client, mgmt_ip, vip, node_user, node_password,
+            vm_credential["user"], vm_credential["password"], "ls")
+
+        stdout = _stdout.read().decode('ascii').strip("\n")
+
+        assert stdout.find("bin") == 0, (
+            f"Failed to ssh to VM management IP {mgmt_ip} "
+            f"on management interface from Harvester node")
+
+        # Check should not SSH to management ip address from external host
+        command = ['/usr/bin/ssh', '-o', 'ConnectTimeout=5', mgmt_ip]
+        try:
+            result = subprocess.check_output(
+                command, stderr=subprocess.STDOUT, shell=False, encoding="utf-8")
+        except subprocess.CalledProcessError as e:
+            result = e.output
+
+        assert "connect to host {0} port 22: No route to host".format(mgmt_ip) in result, (
+            f"Failed: Should not be able to SSH to VM management IP {mgmt_ip}"
+            f"on management interface from Harvester node")
+
+        # cleanup VM
+        delete_vm(api_client, unique_name, wait_timeout)
+
+        # Revert AllowTcpForwarding for ssh jumpstart
```

Comment:
The reversion should always be executed which means it should be placed in a `finally` clause.